### PR TITLE
CO-648: Add JSON pretty printing.

### PR DESCRIPTION
**Pull request by magical**
Add ?pretty=true to any request to get a pretty-printed JSON response.
Even works for error pages, which is nice.

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public void filter(
+            ContainerRequestContext requestContext,
+            ContainerResponseContext responseContext) throws IOException {
+
+        MultivaluedMap<String, String> queryParams =
+            requestContext.getUriInfo().getQueryParameters()
+
+        if (queryParams.getFirst('pretty') == 'true') {
```
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy]: How about if we make the true case insensitive?

Comment (magical) [reply src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy]: Is there a compelling reason to do so?

Comment (floppyears) [reply src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy]: When I think of a developer having to remember or think of the difference between True or true in the url doesn't make sense. We don't add a lot of value to the API by making the 'true' case sensitive. For example, in the directory & locations api I wouldn't want to have to think about the case sensitivity of the text input.

Comment (magical) [reply src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy]: Ignoring case in a search parameter makes sense. Pretty printing is a developer feature—developers are used to details like case mattering. Who would we really be serving by making true case insensitive? I don't think remembering that "true" is lowercase is any harder than remembering that correct spelling is "true", not "yes" or "1" or "t". How far should we go? What about the other parts of the url? You wouldn't expect `/API/V1/Locations` to work, would you? Or `/api/v1/locations?PRETTY=true`? Your argument applies just as well there.

Comment (magical) [reply src/main/groovy/edu/oregonstate/mist/api/PrettyPrintResponseFilter.groovy]: I can tell i'm not convincing you :) Done.

**Pull request merged**